packages/xo-web/src/common/render-xo-item.js

```diff
@@ -17,6 +20,112 @@ const OBJECT_TYPE_TO_ICON = {
   network: 'network',
 }
 
+const COMMON_PROP_TYPES = {
+  link: PropTypes.bool,
+}
+
+const XoItem = ({ children, item, link, to }) =>
+  item !== undefined ? (
+    link ? (
+      <Link to={to} target='_blank'>
+        {children()}
+      </Link>
+    ) : (
+      children()
+    )
+  ) : (
+    <span className='text-muted'>{_('errorNoSuchItem')}</span>
+  )
+
+XoItem.propTypes = {
+  ...COMMON_PROP_TYPES,
+  item: PropTypes.object,
+  to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+}
+// ===================================================================
+
+const XO_ITEM_PROP_TYPES = {
+  ...COMMON_PROP_TYPES,
+  id: PropTypes.string.isRequired,
+}
+
+export const VmItem = [
+  connectStore(() => {
+    const getVm = createGetObject()
+    return {
+      vm: getVm,
+      container: createGetObject(
+        createSelector(getVm, vm => get(() => vm.$container))
+      ),
+    }
+  }),
+  ({ vm, container, ...props }) => (
+    <XoItem item={vm} to={`/vms/${vm.id}`} {...props}>
```

Comment:
`vm` may be undefined.

Reply:
Yes, this case is handled by `XoItem`.

Reply:
Not for the `vm.id` right here which will break with "cannot read property id of undefined".